src/IR/IR.ts

```diff
@@ -18,7 +18,16 @@ export type Expr =
  */
 export type Program = {
   type: "Program";
-  block: Block;
+  imports: Import[];
+  varDeclarations: VarDeclaration[];
+  block: Block;  
+};
+
+/**
+ * Program input (array of strings).
+ */
+ export type Argv = {
```

Comment:
Argv doesn't seem to be referenced anywhere. Should it be directly on Program?

I'm guessing that it would be used to enable certain optimizations such as assuming argc=1 in C programs.

Reply:
> Argv doesn't seem to be referenced anywhere. Should it be directly on Program?
> 
> I'm guessing that it would be used to enable certain optimizations such as assuming argc=1 in C programs.

It represents a nullary builtin - the program input. It will translate to `sys.argv` for Python for example.

Reply:
Oh! That makes sense! It still needs to be added to the `Expr` union though